src/notifications.c

```diff
@@ -338,8 +339,61 @@ void ShardingEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent,
   }
 }
 
-static bool in_asm_trim = false;
-static bool in_asm_import = false;
+//We still do not rely on the TIMER_ID being 0 to check initialization state.
+#define UNITIALIZED_TIMER_ID 0
+
+struct TrimmingDelayCtx {
+  bool checkTrimmingStateTimerIdScheduled;
+  bool enableTrimmingTimerIdScheduled;
+  RedisModuleTimerID checkTrimmingStateTimerId;
+  RedisModuleTimerID enableTrimmingTimerId;
+};
+
+static struct TrimmingDelayCtx trimmingDelayCtx = {
+  .checkTrimmingStateTimerIdScheduled = false,
+  .enableTrimmingTimerIdScheduled = false,
+  .checkTrimmingStateTimerId = UNITIALIZED_TIMER_ID,
+  .enableTrimmingTimerId = UNITIALIZED_TIMER_ID
+};
+
+static void checkTrimmingStateCallback(RedisModuleCtx *ctx, void *privdata) {
+  REDISMODULE_NOT_USED(privdata);
+  // 1. Check counter of queries with old version
+  // 2. If counter is 0, enable trimming and stop enableTrimmingTimer.
+  // 3. Otherwise, reschedule the timer after TRIMMING_STATE_CHECK_DELAY.
+
+  RedisModule_Log(ctx, "verbose", "Checking if we can start trimming migrated slots.");
+  if (ASM_CanStartTrimming()) {
+    RedisModule_Log(ctx, "verbose", "No queries using the old version, Enabling trimming.");

```

Comment:
Let's make this one `notice`, so we have an indication that we enabled trimming before we see the `RedisModule_Log(RSDummyContext, "notice", "Got ASM trim started event.");` line